ethclient/ethclient.go

```diff
@@ -325,13 +325,20 @@ func (ec *Client) SubscribeNewHead(ctx context.Context, ch chan<- *types.Header)
 	return ec.c.EthSubscribe(ctx, ch, "newHeads")
 }
 
-// GetBlockResultByHash returns the blockResult.
+// GetBlockResultByHash returns the blockResult by hash.
 func (ec *Client) GetBlockResultByHash(ctx context.Context, blockHash common.Hash) (*types.BlockResult, error) {
-	var blockResult types.BlockResult
-	if err := ec.c.CallContext(ctx, &blockResult, "eth_getBlockResultByHash", blockHash); err != nil {
-		return nil, err
+	blockResult := &types.BlockResult{}
+	err := ec.c.CallContext(ctx, &blockResult, "eth_getBlockResultByHash", blockHash)
+	if err != nil {
+		err = ec.c.CallContext(ctx, &blockResult, "eth_getBlockResultByNumberOrHash", blockHash)
 	}
-	return &blockResult, nil
+	return blockResult, err
+}
+
+// GetBlockResultByNumber return the blockResult by number.
+func (ec *Client) GetBlockResultByNumber(ctx context.Context, number *big.Int) (*types.BlockResult, error) {
+	blockResult := &types.BlockResult{}
+	return blockResult, ec.c.CallContext(ctx, &blockResult, "eth_getBlockResultByNumberOrHash", toBlockNumArg(number))
 }
```

Comment:
I think this API name is a little confusing.

Reply:
Confused in what way? Or do you have a suitable name?

Reply:
I think we can change this name from `GetBlockResultByNumber`  to `GetBlockResultByNumberOrHash `

Reply:
In eth/tracers/api_blockResult.go already have the api.

Reply:
what about 

`GetBlockResultByNumberOrHash` in eth/tracers/api_blockResult.go
and
`BlockResultByNumberOrHash` here?